### ISSUE TITLE
Regresssion: WASM fails to load due to _scriptDir not defined

### ISSUE THREAD

**Issue by vladmandic** [tensorflow/tfjs#4539]
This is a regression in `tfjs` 2.8.4 compared to 2.8.3  

Setting backend to `wasm` on the main thread results in error:
```
worker.js onmessage() captured an uncaught exception: ReferenceError: _scriptDir is not defined
  threadPrintErr @ e67a52fd-9fdd-4d80-87b0-5a39d7cd567d:1
  onmessage @ e67a52fd-9fdd-4d80-87b0-5a39d7cd567d:1

ReferenceError: _scriptDir is not defined
  WasmBackendModuleThreadedSimd (a0196e5e-0cae-4d78-90e3-7f0714096dfc:1)
  onmessage (e67a52fd-9fdd-4d80-87b0-5a39d7cd567d:1)
```

No issues running TFJS 2.8.3 or older.

This is likely related to pull request #4528

**Comment by jinjingforever** [comment]
Hi Vladimir, 

Thanks for reporting! This is a little bit strange because we did test it in 2.8.4. See this codepin: https://codepen.io/jing-jin/pen/BaLqJoE. If you click "Settings", you can see that the program is loading tfjs core and wasm backend @ 2.8.4.

Could you please provide more context/sample code about this issue? Thanks!

**Comment by vladmandic** [comment]
Ok, this is related to both #4392 and #4528.  

Originally `tfjs-backend-wasm` was not fully minification safe when running through a bundler - minify whitespace and minify syntax was ok, but minify identifiers was not. Then #4528 fixed it and everything was fine for a while.

Now #4392 broke it once more and it's even worse since not even minify syntax works anymore.  
So yes, it's a regression.  

If you want to see a simple project, it's at <https://github.com/vladmandic/stocks> and bundler runs via 'server/build.js'

**Comment by vladmandic** [comment]
closing as fixed in tfjs 2.8.5.  
thank you!
